OpenTrons/opentrons-api
# feat(api-client, app, react-api-client) hook up E-stop modals to desktop app and touchscreen app

# Overview
hook up estop modals to desktop app and touchscreen app



# Test Plan
Test needs a Flex since we don't have E-stop that is capable of connecting to a dev kit
[Desktop app]
pressed modal
1. Open the app and go to Device Details
2. Press E-stop button
Check: the E-stop pressed modal shows up
3. Twist E-stop button
Check: Resume operation button in the E-stop pressed modal is activated (This will take some time) fetching interval is 10sec
4. Click close icon
Check: Device Details page shows E-stop Banner and access to /Devices/:robotName/something (RobotSettings, Protocol Run page) and check those page don't show the E-stop pressed modal
5. Click `Resume operation` button in E-stop Banner
Check: E-stop pressed modal shows up
6. Click `Resume operation` button in E-stop pressed modal

missing modal
1. Open the app and go to Device Details
2. Detatch E-stop from a Flex
Check: the E-stop missing modal shows up
3. Click close icon
Check: Device Details page shows E-stop Banner and access to /Devices/:robotName/something (RobotSettings, Protocol Run page) and check those page don't show the E-stop pressed modal
5. Click text button in the Banner
Check: E-stop missing modal shows up
6. Connect E-stop button to the Flex
Check: E-stop missing modal is dismissed without any actions

[ODD app]
pressed modal
1. Press E-stop button
Check: the E-stop pressed modal shows up
2. Twist E-stop button
Check: Resume operation button in the E-stop pressed modal is activated (This will take some time) fetching interval is 10sec

missing modal
1. Detatch E-stop from a Flex
Check: E-stop missing modal shows up
2. Connect E-stop to the Flex
Check: E-stop missing modal is dismissed


# Changelog
- Add takeover component and react context component 
- Update E-stop modals
- Update each top component to hook up modals and banners to each app
- Update DeviceDetails component and ProtocolRunHeader component



# Review requests
Please see Test plan


# Risk assessment
low-mid
